### PR TITLE
feat(enrichment): add scaffold category, scaffolding engine, and scaffold-incomplete rule

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Adds the section scaffolding engine for `docvet fix` and the `scaffold-incomplete` enrichment rule, completing Story 32.2. Previously, developers had to write missing docstring sections from scratch; now `scaffold_missing_sections()` inserts placeholder sections from AST analysis, and the new rule detects unfilled `[TODO: ...]` markers to block CI until sections are filled in.

- Add `scaffold_missing_sections()` engine in `src/docvet/checks/fix.py` with AST re-extraction for section-specific placeholders (exception names, return types, class fields), one-liner expansion, multi-line insertion, quote/newline preservation, reverse-order processing, and idempotency
- Extend `Finding.category` Literal to include `"scaffold"` with cyan terminal output, medium JSON severity, and summary count formatting across terminal/markdown/JSON reporters
- Add `_check_scaffold_incomplete()` enrichment rule in `_late_rules.py` with regex `[TODO: ...]` detection, section-context extraction for actionable messages, and `scaffold_incomplete` config toggle on `EnrichmentConfig`
- Add `scaffold-incomplete` to `_VALID_ENRICHMENT_KEYS` for pyproject.toml configuration support
- Create `docs/site/rules/scaffold-incomplete.md` rule reference page, update enrichment check and configuration docs
- Add 54 tests (26 unit for scaffolding engine, 11 unit for scaffold-incomplete rule, 11 reporting, 4 integration roundtrip, 2 edge case)

Test: `uv run pytest` — 1714 passed; `uv run docvet check --all` — zero findings

feat(enrichment): add scaffold-incomplete enrichment rule
docs(enrichment): add scaffold-incomplete rule page and config docs
---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [x] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- `src/docvet/checks/fix.py` — new scaffolding engine (394 lines), core logic for section insertion
- `src/docvet/checks/enrichment/_late_rules.py:385-485` — `_check_scaffold_incomplete()` TODO marker detection
- `src/docvet/config/__init__.py:351` — `scaffold-incomplete` in `_VALID_ENRICHMENT_KEYS`
- Idempotency and determinism guarantees (AC 6, 7)
- Integration roundtrip: enrichment → scaffold → re-check → fill → zero findings (AC 9)

### Related
- Story 32.1 spike decision doc (feasibility + POC)
- Epic 32: Quick Wins — Lifecycle & Visibility

### CODE REVIEW

File: src/docvet/checks/fix.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+        first_q = content.find(q)
+        last_q = content.rfind(q)
+        if first_q != -1 and last_q != -1 and first_q != last_q:
+            prefix = content[:first_q]
+            summary = content[first_q + 3 : last_q].strip()
+            new_lines = [f"{prefix}{q}{summary}{newline}"]
+            for section in sections:
+                new_lines.append(newline)
+                new_lines.extend(_build_section_lines(section, node, indent, newline))
+            new_lines.append(f"{indent}{q}{newline}")
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/docvet/checks/fix.py]: Keeping as-is — trailing content after one-liner closing quotes (e.g. `# comment`, `; stmt`) doesn't occur in ruff-clean codebases. Semicolons are blocked by E702, and `# type: ignore`/`# noqa` aren't used on docstring lines. Noted as a known limitation for exotic inputs.

File: src/docvet/checks/fix.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+    "missing-returns": "Returns",
+}
+
+# Section header pattern for idempotency (existing section detection).
+_SECTION_PATTERN = re.compile(
+    r"^\s*(Args|Returns|Raises|Yields|Receives|Warns"
+    r"|Other Parameters|Attributes|Examples|See Also"
+    r"|Notes|References|Warnings|Extended Summary|Methods):\s*$",
+    re.MULTILINE,
+)
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/docvet/checks/fix.py]: Keeping as-is — the section header list is defined by the Google Style Guide and has been frozen across 21 rules. Importing the regex from enrichment would couple fix.py to an internal implementation detail. Two copies of a frozen constant is acceptable defensive decoupling. If a 16th header type is ever added, grep-and-update is a 30-second task.
